src/adapter/sources.ts

```diff
@@ -355,8 +357,29 @@ export class SourceContainer {
   }
 
   source(ref: Dap.Source): Source | undefined {
-    if (ref.sourceReference) return this._sourceByReference.get(ref.sourceReference);
-    if (ref.path) return this._sourceByAbsolutePath.get(ref.path);
+    if (ref.sourceReference) {
+      const byRef = this._sourceByReference.get(ref.sourceReference);
```

Comment:
What's going on here?

Reply:
By default breakpoints set before reload will have a source reference, but it's invalid. If we saw that, we used to return undefined, this change falls back to looking at the path/adapter data when possible.

Reply:
I think I misread it, it might be more clear to return `byRef` instead of calling get a second time.